handlers/handlers.go

```diff
@@ -40,68 +42,98 @@ func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
+//GeographyHomepageRender ...
+func GeographyHomepageRender(rend RenderClient) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
 		var page geographyHomepage.Page
 
-		resp, err := http.Get(`https://api.dev.cmd.onsdigital.co.uk/v1/code-lists` + localAuthority)
+		homePageLink := url + `/code-lists` + localAuthority
+
+		resp, err := http.Get(homePageLink)
 		if err != nil {
+			log.Error(err, log.Data{"error getting data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": err})
+			log.Error(err, log.Data{"test": "error http.Get"})
 			setStatusCode(req, w, err)
 			return
 		}
+		if resp.StatusCode == 404 {
+			log.Error(err, log.Data{"error getting data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": resp.StatusCode})
+			setStatusCode404(req, w, err)
+			return
+		}
 		b, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
+			log.Error(err, log.Data{"error getting the .Body data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender .Body returned ": resp.Body})
 			setStatusCode(req, w, err)
 			return
 		}
 		var codelistresults models.CodeListResults
 		err = json.Unmarshal(b, &codelistresults)
 		if err != nil {
+			log.Error(err, log.Data{"error Unmarshaling the .Body data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": err})
 			setStatusCode(req, w, err)
 			return
 		}
 		var codelist models.CodeList
 		err = json.Unmarshal(b, &codelist)
 		if err != nil {
+			log.Error(err, log.Data{"error Unmarshaling the .Body data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": err})
 			setStatusCode(req, w, err)
 			return
 		}
 
-		geographyTypes := ""
+		var geographyTypes []geographyHomepage.Items
+		geographyTypesLabel := ""
+		geographyTypesID := ""
 		for i := range codelistresults.Items {
-			log.Debug("test", log.Data{
+			log.Debug("for loop", log.Data{
 				"geographyTypesTest": codelistresults.Items[i],
 			})
 
-			geographyTypes = geographyTypes + codelistresults.Items[i].Name
-
-		}
-
-		page.Data.AreaTypes = []geographyHomepage.AreaType{
-			{Name: "Countries"},
-			{Name: "Regions"},
-			{Name: geographyTypes},
+			geographyTypesID = codelistresults.Items[i].Links.Self.ID
+
+			resp, err := http.Get(codelistresults.Items[i].Links.Editions.Href)
+			if err != nil {
+				setStatusCode(req, w, err)
+				return
+			}
+			b, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				setStatusCode(req, w, err)
+				return
+			}
+			var codelistresults models.CodeListResults
+			err = json.Unmarshal(b, &codelistresults)
+			if err != nil {
+				setStatusCode(req, w, err)
+				return
+			}
+			geographyTypesLabel = codelistresults.Items[0].Label
+			geographyTypes = append(geographyTypes, geographyHomepage.Items{Label: geographyTypesLabel, ID: geographyTypesID})
+			log.Debug("for loop", log.Data{
```

Comment:
same for this log - is it really needed? If so can you update it with some context and details about the state.